R/lm_imp.R

```diff
@@ -0,0 +1,216 @@
+#' @inherit step_center return
+#' @param ... One or more selector functions to choose variables. For
+#' `step_lmimpute`, this indicates the variables to be imputed; these variables
+#' **must** be of type `numeric`. When used with `imp_vars`, the dots indicates
+#' which variables are used to predict the missing data in each variable. See
+#' [selections()] for more details. For the `tidy` method, these are not
+#' currently used.
+#' @param role Not used by this step since no new variables are created.
+#' @param impute_with A call to `imp_vars` to specify which variables are used
+#'  to impute the variables that can include specific variable names separated
+#'  by commas or different selectors (see [selections()]). If a column is
+#'  included in both lists to be imputed and to be an imputation predictor, it
+#'  will be removed from the latter and not used to impute itself.
+#' @param models The [lm()] objects are stored here once the linear models
+#'  have been trained by [prep.recipe()].
+#' @return An updated version of `recipe` with the new step added to the
+#'  sequence of existing steps (if any). For the `tidy` method, a tibble with
+#'  columns `terms` (the selectors or variables selected) and `model` (the
+#'  bagged tree object).
+#' @keywords datagen
+#' @concept preprocessing
+#' @concept imputation
+#' @export
+#' @details For each variable requiring imputation, a linear model is fit
+#'  where the outcome is the variable of interest and the predictors are any
+#'  other variables listed in the `impute_with` formula. The variable(s) to be
+#'  imputed must be of type `numeric`; furthermore, they will be coerced to type
+#'  `double` as the missing values will be set as the results of a `predict.lm`.
+#'
+#'  Note that if a variable that is to be imputed is also in `impute_with`,
+#'  this variable will be ignored.
+#'
+#'  Since this is a regression, the imputation has been set to throw an error if any of
+#'  the predictors used for imputing contain missing values.
+#'
+#' @references Kuhn, M. and Johnson, K. (2013). *Applied Predictive Modeling*.
+#'  Springer Verlag.
+#' @examples
+#' library(modeldata)
+#' data("ames")
+#'
+#' ## assign all 0s in Lot_Frontage to be NA
+#' ames_missing <- ames
+#' ames_missing[ames$Lot_Frontage == 0, ]$Lot_Frontage <- NA
+#'
+#' rec <- recipe(head(ames_missing))
+#' \dontrun{
+#' imputed_ames <- rec %>%
+#'   step_lmimpute(Lot_Frontage, impute_with = imp_vars(Lot_Area)) %>%
+#'   prep(ames_missing) %>%
+#'   bake(ames_missing)
+#'
+#' sum(is.na(ames_missing[ames$Lot_Frontage == 0, ] ))
+#' sum(is.na(imputed_ames[ames$Lot_Frontage == 0, ] ))
+#'
+#' ## show that the NAs (originally 0) are now imputed linearly
+#' par(mfrow=c(2, 1))
+#' plot(ames$Lot_Area, ames$Lot_Frontage, col = ifelse(ames$Lot_Frontage == 0, "red", "black"))
+#' plot(imputed_ames$Lot_Area, imputed_ames$Lot_Frontage, col = ifelse(is.na(ames_missing$Lot_Frontage), "red", "black"))
+#' }
+
+step_lmimpute <-
```

Comment:
Should this function be named `step_impute_lm()`, rather than `step_lm_impute()`? I know it slightly deviates from the current naming system (e.g. `step_knnimpute()`), but it follows the `tidy`-ish approach of putting the common text first to make fuzzy-narrowing and tab auto-completion work better (and easier to remember!).

Reply:
In the future, if you decide to soft-deprecate the naming style `step_knnimpute` _et al._, then I think it creates a cleaner API for the impute steps to follow the `step_impute_*` naming system.

Reply:
@topepo Do you agree? Easy change for me.

This PR isn't dead; I will incorporate the changes, it's just a crazy time in my life. Will hopefully have some time soon to tie this off.

Reply:
No worries. I may do a release soon in which can I will merge and make some modifications.

Reply:
@mattwarkentin I agree with the name suggestion for this function. Can you add another issue to "harmonize" the names of the existing steps?

Reply:
Sure, no problem. Will post that issue shortly.